ui/widgets/calendar.js

```diff
@@ -232,146 +239,157 @@ return $.widget( "ui.calendar", {
 	},
 
 	_createCalendar: function() {
-		var classes = "ui-calendar ui-widget ui-widget-content ui-helper-clearfix ui-corner-all",
-			pickerHtml = this._buildHeaderButtons();
+		this.element
+			.attr( "role", "region" )
+			.append( this._buildHeaderButtons() );
 
 		if ( this.options.numberOfMonths === 1 ) {
-			pickerHtml += this._buildHeader() + this._buildGrid();
-			this.element.attr( "aria-labelledby", this.gridId + "-title" );
+			this.element
+				.attr( "aria-labelledby", this.gridId + "-title" )
+				.append( this._buildHeader() )
+				.append( this._buildGrid() );
 		} else {
-			pickerHtml += this._buildMultiplePicker();
-			classes += " ui-calendar-multi";
+			this._buildMultiplePicker();
 		}
 
-		this.element
-			.addClass( classes )
-			.attr( "role", "region" )
-			.html( pickerHtml );
+		this._addClass( this.element, "ui-calendar", "ui-widget ui-widget-content ui-helper-clearfix" );
 
-		this.prevButton = this.element.find( ".ui-calendar-prev" );
-		this.nextButton = this.element.find( ".ui-calendar-next" );
 		this._refreshHeaderButtons();
-
 		this._createButtonPane();
 
 		this.grid = this.element.find( ".ui-calendar-calendar" );
 	},
 
 	_buildMultiplePicker: function() {
-		var headerClass,
-			html = "",
+		var html, header,
+			rowBreak = $( "<div>" ),
 			currentDate = this.viewDate,
 			months = this.viewDate.months( this.options.numberOfMonths - 1 ),
-			labelledby = [],
+			labelledBy = [],
 			i = 0;
 
 		for ( ; i < months.length; i++ ) {
 
 			// TODO: Shouldn't we pass date as a parameter to build* fns instead of setting this.date?
 			this.viewDate = months[ i ];
 			this.gridId = this.id + "-" + i;
+			labelledBy.push( this.gridId + "-title" );
 
-			labelledby.push( this.gridId + "-title" );
-			headerClass = "ui-calendar-header ui-widget-header ui-helper-clearfix";
+			html = $( "<div>" );
+			this._addClass( html, "ui-calendar-group" );
+
+			header = this._buildHeader();
+			this._removeClass( header, "ui-corner-all" );
```

Comment:
You'll need to use different classes for single and multi-picker instances. This will allow the default values for those to apply the correct corner classes. See `ui-selectmenu-button-open` and `ui-selectmenu-button-closed` for a similar example.